romancal/pipeline/exposure_pipeline.py

```diff
@@ -151,7 +151,7 @@ def process(self, input):
             if result.meta.exposure.type == "WFI_IMAGE":
                 result = self.flatfield(result)
                 result = self.photom(result)
-                result = self.source_detection(result)
```

Comment:
I tried
```python
result = self.source_catalog(result, return_updated_model=True)
```
but it fails (keyword is not known). I'll check with @ddavis-stsci to see if we can do something here.

Reply:
Since it's part of the `spec` it's expected in the `Step.__init__` (and a few other places). The `__call__` is an [alias](https://github.com/spacetelescope/stpipe/blob/01f1956f3b3c3750a26790d5217cf909cad4b95a/src/stpipe/step.py#L571) to `run` which [only supports](https://github.com/spacetelescope/stpipe/blob/01f1956f3b3c3750a26790d5217cf909cad4b95a/src/stpipe/step.py#L411) `*args`.

Perhaps `return_updated_model` could be a `Step` attribute, set in the pipeline with:
```
self.source_detection.return_updated_model = True
```
and checked in the `Step.process` with:
```
if getattr(self, "return_updated_model, False):
```

Reply:
Thanks for the suggestion, @braingram! Let me give it a shot.

Reply:
Looks like this approach worked, @braingram!